content/blog/zephyr-tracing-and-profiling/index.md

```diff
@@ -0,0 +1,279 @@
+
+- CPU Usage
+- Execution times
+- Response times
+- Periodicity
+- Separation
+- Fragmentation
+
+These statistics can help an experienced developer spot problems with any of the subsystems, as well as check timing and CPU load under any execution regime. Any anomaly can lead to a detailed investigation on what's hindering performance, including glitches that will only be visible when enough data is collected.
+
+Another interesting capability of Tracealyzer is helping monitor heap utilization. The following graph shows memory allocation/de-allocation over time, and it is paired with the synchronized *trace view* right below it. This renders possible to see if memory is allocated and freed accordingly or if memory leaks might be happening.
+
+<a href="img/img005.webp" target="_blank">
+    {{< figure
+        default=true
+        src="img/img005.webp"
+        alt=""
+        caption="Memory heap trace example"
+    >}}
+</a>
+
+If memory leaks were present, allocation would only increase with time instead of reaching a mean usage level.
+
+Another example of use case is to analyze the interaction between threads and shared resources such as message queues. Spotting problems like producers-consumers can be difficult, specially if it's a glitch. If a synchronized signal can be added to the acquisition to help pointing to the timestamp where the problem is happening - say, when a symptom is visible to the outside world - the acquired data can be very useful to solve the problem.
+
+<a href="img/img006.webp" target="_blank">
+    {{< figure
+        default=true
+        src="img/img006.webp"
+        alt=""
+        caption="Producer-consumer example"
+    >}}
+</a>
+
+The examples below show custom channels added to capture application data:
+
+<a href="img/img007.webp" target="_blank">
+    {{< figure
+        default=true
+        src="img/img007.webp"
+        alt=""
+        caption="Custom channel state transitions"
+    >}}
+</a>
+
+<a href="img/img008.webp" target="_blank">
+    {{< figure
+        default=true
+        src="img/img008.webp"
+        alt=""
+        caption="Custom channel signals"
+    >}}
+</a>
+
+## Final remarks
+
+While these examples are generic and the amount of possibilities that *tracing* and *profiling* tools can offer is vast, this article is meant to just scratch the surface and share some insights on how powerful these tools can be. Even though they may be more applicable to analyze complex systems, it is well worth knowing them, as they may happen to be just the right tool for a specific problem. Besides that, even for use cases that require high reliability, using them with the intent of *profiling* the system in order to check how it behaves over time and certify its performance can be probably the best approach, as you'll have real data to back up the perception that it is performing as intended.

```

Comment:
This paragraph is pretty long. Suggest splitting it into two. The first paragraphs briefly mentions the scope of this article, while the second one reiterates how knowing the information presented in this article can be helpful.

```suggestion
While these examples are generic and the amount of possibilities that *tracing* and *profiling* tools can offer is vast, this article just scratches the surface in sharing some insights on how powerful these tools can be.

Even though these examples may be more applicable to analyzing complex systems, it is well worth knowing them, as they may happen to be just the right tool for a specific problem. Besides that, even for use cases that require high reliability, using them with the intent of *profiling* the system in order to check how it behaves over time and certify its performance can be probably the best approach, as you'll have real data to back up the perception that it is performing as intended.

```